File: 2024-2/ejercicios/poo-sobrecarga-de-operadores/cafeteria/Ingrediente.cpp
/* 
 * File:   Ingrediente.cpp
 * Author: erichuiza
 * 
 * Created on October 7, 2024, 12:42 PM
 */

#include "Ingrediente.h"

Ingrediente::Ingrediente() {
}

Ingrediente::Ingrediente(const Ingrediente& orig) {
    *this = orig;
}

Insumo* Ingrediente::getInsumo() const {
    return insumo;
}

char* Ingrediente::getUnidadMedida() const {
    return unidadMedida;
}

double Ingrediente::getCantidad() const {
    return cantidad;
}

void Ingrediente::setInsumo(Insumo* insumo) {
    this->insumo = insumo;
}

void Ingrediente::setUnidadMedida(const char* unidadMedida) {
    this->unidadMedida = new char[strlen(unidadMedida) + 1];
    strcpy(this->unidadMedida, unidadMedida);
}
    
void Ingrediente::setCantidad(const double cantidad) {
    this->cantidad = cantidad;
}

void Ingrediente::operator=(const Ingrediente& ingrediente) {
    this->setInsumo(ingrediente.getInsumo());
    this->setUnidadMedida(ingrediente.getUnidadMedida());
    this->setCantidad(ingrediente.getCantidad());
}

Ingrediente::~Ingrediente() {
    delete[] unidadMedida;
}

File: 2024-2/sesiones-de-clase/sesion-de-clase-23/src/Ingrediente.java
/**
 *
 * @author erichuiza
 */
class Ingrediente {
    private double cantidad;
    private String unidadMedida;
    private Insumo insumo;

    public double getCantidad() {
        return cantidad;
    }

    public String getUnidadMedida() {
        return unidadMedida;
    }

    public Insumo getInsumo() {
        return insumo;
    }
    
    public void setCantidad(final double cantidad) {
        this.cantidad = cantidad;
    }

    public void setUnidadMedida(final String unidadMedida) {
        this.unidadMedida = unidadMedida;
    }

    public void setInsumo(final Insumo insumo) {
        this.insumo = insumo;
    }
}